backend/app/routers/map.py

```
@@ -30,34 +33,48 @@ def get_db():
+                    popup=description,
+                    color='red',
+                    opacity=0.7,
+                    radius=7,
+                )
+            )
+        user_map.add_child(fg)
+
+    try:
+        user_map.save(str(map_path))
+    except Exception as e:
+        raise HTTPException(status_code=500, detail="Map generation failed")
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)